### DISCUSSION TITLE
How can I convert a RTF/CRE stroke string to Stroke object?

### DISCUSSION THREAD

**Discussion by user202729** [opensteno/plover#1224] [Plugin Support] [answered]
\# Do we have tags?

The question is in the title.

**Comment by user202729** [comment]
I don't think there's a function in Plover.

Method 1. Use Benoit's new library [plover-stroke](https://github.com/benoit-pierre/plover_stroke). See the README for more documentation.

Method 2. Use `steno_to_stroke` function in testing build utils (`plover_build_utils.testing.steno_to_stroke`).

**Reply by benoit-pierre** [reply]
Tags?

**Reply by user202729** [reply]
I mean, perhaps there can be some way to classify the discussion with some popular topic.

Not sure if GitHub discussion have such a feature.

**Reply by benoit-pierre** [reply]
Categories (top-right)?

**Reply by user202729** [reply]
I was kind-of thinking about stack overflow's tag thing. This is okay, but not as specific.

**Reply by panathea** [reply]
https://github.com/github/feedback/discussions/19

**Comment by benoit-pierre** [comment, marked as answer]
Anyway, I think `plover_stroke` is more robust than `plover_build_utils.testing.steno_to_stroke`. It also does not rely on `plover.system`.

**Comment by user202729** [comment]
Side note:

To use `plover_stroke` in Plover plugins: (the set & is required in some cases, see https://github.com/openstenoproject/plover/discussions/1449)

```python
from plover import system
from plover_stroke import BaseStroke
class Stroke_(BaseStroke): pass
Stroke_.setup(system.KEYS, {*system.IMPLICIT_HYPHEN_KEYS} & {*system.KEYS}, system.NUMBER_KEY, system.NUMBERS)

stroke_=Stroke_("STK")
```

Or in an independent script: (without the overhead mentioned in #1292)

```python
from plover.system import english_stenotype as e
from plover_stroke import BaseStroke
class Stroke_(BaseStroke): pass
Stroke_.setup(e. ... )
```

Sometimes you would need to convert `Stroke_` object to `Stroke` object too:

```python
from plover.steno import Stroke
print(Stroke(stroke_.keys()))
```



Note that if you use this in a Python dictionary, the `Stroke_` constructor will raise `ValueError` if the RTF/CRE string is invalid (which can happen if you enter an invalid stroke into the "*Strokes:*" text box in the "*Add translation*" dialog). Handle it correspondingly.
